stdlib/public/core/Result.swift

```diff
@@ -89,6 +89,21 @@ public enum Result<Success, Failure: Error> {
   /// Returns a new result, mapping any success value using the given
   /// transformation and unwrapping the produced result.
   ///
+  /// Use this method when you need to transform the value of a `Result`
+  /// instance eventhough it produces a nested result type.
```

Comment:
This sentence doesn't read clearly to me. What would you think of this as an alternative? It would be fine to join this paragraph with the next, as well.

> Use this method when your transformation produces another `Result` type.

Reply:
That sentence read clearer. I'll change it to your sentence.